go/l2geth-exporter/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"math/big"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/go/l2geth-exporter/l1contracts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func main() {
+	listenAddress := os.Getenv("LISTEN_ADDRESS")
+	if listenAddress == "" {
+		listenAddress = ":9100"
+	}
+
+	log.Root().SetHandler(log.CallerFileHandler(log.StdoutHandler))
+
+	l1Url := os.Getenv("L1_URL")
+	if l1Url == "" {
+		log.Error("L1_URL environmental variable is required")
+		os.Exit(1)
+	}
+	ctcAddress := os.Getenv("CTC_ADDRESS")
+	if ctcAddress == "" {
+		log.Error("CTC_ADDRESS environmental variable is required")
+		os.Exit(1)
+	}
+	client, err := ethclient.Dial(l1Url)
+	if err != nil {
+		log.Error("Problem connecting to L1: %s", err)
```

Comment:
Missing `os.Exit(1)`?

Reply:
Incidentally, you can also do `log.Crit` which will print the log line then call `os.Exit(1)` for you.

Reply:
I considered, but decided it should continue to try and update the success metric instead.